lib/render/achievement.php

```diff
@@ -59,14 +38,44 @@ function GetAchievementAndTooltipDiv(
         }
     }
 
-    return "<div class='inline' onmouseover=\"Tip('$tooltip')\" onmouseout=\"UnTip()\" >" .
+    return "<div class='inline' onmouseover=\"Tip(loadCard('achievement', $achID))\" onmouseout=\"UnTip()\" >" .
         "<a href='/achievement/$achID'>" .
         "$smallBadge" .
         "$displayable" .
         "</a>" .
         "</div>";
 }
 
+function renderAchievementCard(int $achievementId): string
+{
+    $achData = [];
+    getAchievementMetadata($achievementId, $achData);
+
+    $badgeName = $achData['BadgeName'];
+    $achNameStr = $achData['AchievementTitle'];
+    $achDescStr = $achData['Description'];
+    $gameNameStr = $achData['GameTitle'];
+    $achPoints = $achData['Points'];
+
+    $tooltip = "<div class='tooltip-body flex items-start gap-2 p-2' style='max-width: 400px'>";
+    $tooltip .= "<img src='" . media_asset("Badge/$badgeName.png") . "' width='64' height='64' />";
+    $tooltip .= "<div>";
+    $tooltip .= "<div><b>$achNameStr</b></div>";
+    $tooltip .= "<div class='mb-1'>$achDescStr</div>";
+    if ($achPoints) {
+        $tooltip .= "<div>$achPoints Points</div>";
+    }
+    $tooltip .= "<div><i>$gameNameStr</i></div>";
+
+    // TODO: extra text should tell if user has unlocked the achievement; use request()->user() for that
+    // $tooltip .= $extraText;
```

Comment:
Were you planning to address this now? It is a change in behavior for the tooltips on a user profile. In addition to not showing the unlock time, it always shows the unlocked image.

Reply:
I feel like i should, yes - it leaves a gap in expected behaviour; yet, I'm not sure if that unlock information grants enough value in all contexts? (It probably does)

Reply:
I think it would be neat to see whether or not I've unlocked an achievement when achievements are referenced in the forums. Although, I'm not sure that the image should be greyed out if the user hasn't unlocked the achievement in that case.

It may be unnecessary in something like the ticket manager, but I don't think it would detract from the functionality.

Perhaps this could be an argument towards pre-generating those tooltips (with the unlock data) on the profile page, and leave the non-profile page requests without the unlock information?

Reply:
Conversely, I've considered putting game progress in the game tooltips (once the denormalized data is ready). This solution makes that even more reasonable, and potentially possible without the denormalized data.
![image](https://user-images.githubusercontent.com/32680403/192183975-dc78a76a-262b-4105-a4bd-93fabc82b26a.png)

Reply:
Also, the unlock state/timestamp information in the tooltip for achievements on user B's profile should be user B's unlock state/timestamp, but the current user's. To do that via the lazyCard would require passing additional context.

Reply:
Agreed. 
That line is more important than I was aware for achievement unlock contexts where it's not shown elsewhere on the page (user pages as you mentioned).

Adding game progress wouldn't be as expensive anymore with similar caching, indeed.

I'll give it a shot right away.

Reply:
> I'm not sure that the image should be greyed out if the user hasn't unlocked the achievement in that case.

I'd stick with always using the colored version for now?

Reply:
> I'd stick with always using the colored version for now?

I think that's fine. It'll be consistent with the dynamically loaded tooltips everywhere else in the code. (i.e. hovering over a locked achievement in the game page will now show the colored version). Although, I'm still leaning towards removing those.

Reply:
You mean removing the tooltips altogether or changing to the grayscale image?

Reply:
https://github.com/RetroAchievements/RAWeb/pull/1142#discussion_r979516891

The tooltip is redundant on the game page and should probably just be removed.

If it's not removed, it'll be loaded dynamically and show the non-gray image, so showing the non-gray image in the tooltip on the profile page would be consistent.

Reply:
I see, I don't think removing the tooltip is necessary. It's potentially cached anyways and keeps it consistent with how all the other resource avatars work; if there's an icon or resource name it can also hold the trigger for fetching the tooltip card. I'd expect it to load the tooltip nonetheless.

Reply:
Oh, wait, I just realized you were referring to a different part of the game page than what I had in mind 😅 Yeah, the achievement tooltips are redundant there.

The methods should be renamed, really, in v2 they were all called avatars.
Adding a parameter to enable or disable tooltips might work fine then. I'll give that a try.

Reply:
Working on a separate PR for the optional tooltip data.